tests/end-to-end/upgrade/pkg/tests/function/function.go

```diff
@@ -0,0 +1,188 @@
+package function
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"time"
+	"strings"
+
+	"github.com/google/uuid"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+
+	kubelessV1 "github.com/kubeless/kubeless/pkg/apis/kubeless/v1beta1"
+	kubeless "github.com/kubeless/kubeless/pkg/client/clientset/versioned"
+	"github.com/sirupsen/logrus"
+)
+
+type FunctionUpgradeTest struct {
+	functionName, uuid string
+	kubelessClient     *kubeless.Clientset
+	coreClient         *kubernetes.Clientset
+}
+
+func NewFunctionUpgradeTest() (*FunctionUpgradeTest) {
+	kubeconfig := os.Getenv("KUBECONFIG")
```

Comment:
kubeconfig is creasted in the main.go. You can follow the patter which is used there and pass to the constructor proper clientset (interface)